addons/sale/tests/test_sale_order.py

```diff
@@ -196,3 +217,160 @@ def test_cost_invoicing(self):
+            'applied_on': '1_product',
+            'product_tmpl_id': self.product_deliver.product_tmpl_id.id,
+            'compute_price': 'percentage',
+            'percent_price': 10
+        })
+
+        # Create an order lines
+        self.SaleOrderLine.create({
+            'order_id': self.sale_order_pricelist.id,
+            'product_id': self.product_order.id
+        })
+        self.SaleOrderLine.create({
+            'order_id': self.sale_order_pricelist.id,
+            'product_id': self.service_deliver.id
+        })
+        self.SaleOrderLine.create({
+            'order_id': self.sale_order_pricelist.id,
+            'product_id': self.service_order.id
+        })
+        self.SaleOrderLine.create({
+            'order_id': self.sale_order_pricelist.id,
+            'product_id': self.product_deliver.id
+        })
+
+        # Check that pricelist of the SO has been applied on the sale order lines or not
+        for line in self.sale_order_pricelist.order_line:
+            if self.sale_order_pricelist.pricelist_id in line.product_id.item_ids.mapped('pricelist_id'):
+                for item in self.sale_order_pricelist.pricelist_id.item_ids.filtered(lambda l: l.product_tmpl_id == line.product_id.product_tmpl_id):
+                    price = item.percent_price
+                    self.assertEquals(price, (line.product_id.list_price - line.price_unit)/line.product_id.list_price*100, 'Pricelist of the SO should be applied on an order line')
+            else:
+                self.assertEquals(line.price_unit, line.product_id.list_price, 'Pricelist of the SO should not be applied on an order line')
+
+    def test_sale_with_pricelist_formulas(self):
+        """ Test SO with the pricelist which one have compute price formula """
+        # Add group 'Discount on Lines' to the user
+        self.env.user.write({'groups_id': [(4, self.env.ref('sale.group_discount_per_so_line').id)]})
+        product_categ = self.env.ref('product.product_category_1')
+
+        # Apply product category on consumable products and also on a peicelist
+        with Form(self.product_order) as product:
+            product.categ_id = product_categ
+        with Form(self.product_deliver) as product:
+            product.categ_id = product_categ
+
+        # Create the pricelist items
+        self.PricelistItem.create({
+            'pricelist_id': self.pricelist_without_disct.id,
+            'applied_on': '2_product_category',
+            'categ_id': product_categ.id,
+            'compute_price': 'formula',
+            'base': 'standard_price',
+            'price_discount': 15
+        })
+        self.PricelistItem.create({
+            'pricelist_id': self.pricelist_with_disct.id,
+            'applied_on': '3_global',
+            'compute_price': 'percentage',
+            'price_discount': 10
+        })
+
+        # Apply pricelist on the SO and create an order lines
+        sale_order = Form(self.sale_order_pricelist)
```

Comment:
no SSF here please.

Reply:
Need to do it using SSF otherwise not getting onchange effect.